lib/IRGen/MetadataRequest.cpp

```diff
@@ -2198,11 +2211,15 @@ emitMetadataAccessByMangledName(IRGenFunction &IGF, CanType type,
   }
 
   // Get or create a shared helper function to do the instantiation.
+  auto instantiationFnName =
+      request.isStaticallyAbstract()
+          ? "__swift_instantiateConcreteTypeFromMangledNameAbstract"
+          : "__swift_instantiateConcreteTypeFromMangledName";
```

Comment:
Thanks for implementing this, Arnold! I wonder whether, if we're targeting Swift >= 5.2, we could only emit `__swift_instantiateConcreteTypeFromMangledNameAbstract` and use it for all metadata requests. That would probably mean a couple more instructions at call sites to materialize the metadata request kind argument, though.